modules/dnn/include/opencv2/dnn/all_layers.hpp

```diff
@@ -1101,6 +1101,16 @@ CV__DNN_INLINE_NS_BEGIN
         static Ptr<LayerNormLayer> create(const LayerParams& params);
     };
 
+    class CV_EXPORTS GemmLayer : public Layer {
+    public:
+        bool trans_a;
+        bool trans_b;
+        float alpha;
+        float beta;
```

Comment:
As OpenCV initializes all the parameters from `LayerParams`, these fields can be moved to the Impl class and became private.

Reply:
My thoughts are parameters in classes in `all_layers.hpp` are from standard layer definitions, which are most probably the same across different frameworks (ONNX, TF, ...). And parameters in impl classes are implementation specific. It is more semantic and meanful to me and I want to keep it in this way.